db/core/ecmcMotorRecord.template

```diff
@@ -35,17 +34,19 @@ record(motor,"$(PREFIX)$(MOTOR_NAME)")
     field(SREV, "$(SREV=1)")
     field(UREV, "$(UREV=1.0)")
     field(VMAX, "$(VMAX=0)")
+    $(ECMC_BLOCK_ARCHIVER=#)info(aa_policy, "default")
+    $(ECMC_BLOCK_ARCHIVER=#)info(archiver, "$(ECMC_ARCHIVER=)")
```

Comment:
I'd argue that there is no harm in having those `info` fields present for _all_ facilities. In case any of them isn't using them, fine. If they are using the same keyword, they can set the MACRO as needed.
This will also remove the need of yet another "BLOCK" MACRO flying around.
```suggestion
    info(aa_policy, "${ECMC_ARCHIVER_POLICY=}")
    info(archiver, "${ECMC_ARCHIVER=}")
```

Reply:
You may want to do this differently then it that case: `aa_policy` and `archiver` are just the key names _we_ have opted for. And it is entirely possible to have `""` for archiver to have a specific meaning - we considered having `""` mean our main technical network, as most PVs we archive will be from there.

Reply:
Is there any harm in having `info` fields that are not being used at a facility?
It's already common for other modules to riddle their templates with autosave/restore fields, that not every facility is using.
Do I understand that correctly? You are giving an empty string a strong meaning ... I guess this design decision could be the start of a long discussion.
I'm perfectly OK to use `${ECMC_ARCHIVER_POLICY=default}` in ecmccfg.

Reply:
My point is that CF-AA archival is available to all, but only at ESS do we use these specific key names for this feature (IIRC the defaults are archive and archiver). So I think it might be simpler to either go with the initial suggestion by @cdn-lnk else to build in a way to also override the keys; I do not think there is any point to only making the value of e.g. aa_policy a macro.

Reply:
With the initial suggestion, you refer to the blocker `$(ECMC_BLOCK_ARCHIVER=#)`?
I'm OK with that.
But, the value for the policy must be a macro.

Reply:
Hi all,

Sorry for the late reply, I got stuck on some test yesterday.

1. Regarding inclusion of `$(ECMC_ARCHIVER_POLICY)`
    I will include that. Makes the code more generic without any penalties to functionality or maintenance.
 
2. Regarding `$(ECMC_BLOCK_ARCHIVER)`
    I would like the default behavior to be NOT archive; to prevent stale PVs on the archiver appliance (imagine every test IOC ever ran being archived by default).
    This can be accomplished by a) commenting out the info tags or b) setting `$(ECMC_ARCHIVER_POLICY)` or `$(ECMC_ARCHIVER)` to some invalid value.
    Since the concept of "invalid input" can differ between facilities and implementations I am tending towards option a).

3. Regarding default values for `${ECMC_ARCHIVER_POLICY=}` and `${ECMC_ARCHIVER=}`
    From an ESS point of view `${ECMC_ARCHIVER_POLICY=default}` and `${ECMC_ARCHIVER=tn}` would be really convenient but may not be ideal if another facility happens to use the same tags.
    Since having empty string may not be a solution (see item 2 above) we could have the default values based on `NAMING` this would however have the inconvenience of expanding the `ecmcEpicsEnvSetCalcTernary` block. A compromise could be create a new [archive.cmd](archive.cmd) file that is called during [startup.cmd](startup.cmd) instead of being called from the [motion/ecmc_axis_mr.cmd](motion/ecmc_axis_mr.cmd)

4. Lastly, I would like to allow archiving only if `NAMING=ESSnaming` that way we don't get "wrong" names.
But, once again, this may not play well if other facilities.
I don't have a clear generalization for this in mind right now but one possible idea would be changing `$(ECMC_BLOCK_ARCHIVER)` to  `$(ECMC_ESS_ARCHIVER)` and each facility would have a macro for its own tags.

@kivel please let me know you thoughts.

Reply:
Hi Carlos
Thanks for the summary!

- I already expressed my concern about giving an empty string a meaning on the backend. But that's out of scope for this discussion.
- If it was only me, I'd not archive anything without an explicit setting in the `startup.iocsh`, any implicit mechanism will lead to accidental archiving. That goes for policy and instance.
- Since the fields might be used by other facilities, the limitation on `NAMING=ESSnaming` is not acceptable, as you pointed out.

So "invalid" field values for policy and instance as default, I'd assume "none" might work here. The blocking is still optional in my mind but I already rolled over in a previous comment.

Keep in mind, those are my two cents from an outside perspective.

Reply:
@kivel ,

I just pushed a new version.

It turned out very similar to your suggestion but I kept `$ECMC_BLOCK_ARCHIVER` and moved all archiver configuration to a dedicated file.

I decided to not included [configureArchiver.cmd](configureArchiver.cmd) into [startup.cmd](startup.cmd) and opted for the integrator calling it from `st.cmd`.

N.B.: I remove `$ECMC_AXISFIELDINIT` as it was not used for anything.

I also fixed some minor bugs, like missing commas.